src/components/Wallet/WalletDeposit.jsx

```diff
@@ -50,47 +53,41 @@ class WalletDeposit extends React.Component {
     clearInterval(depositTimer);
   }
 
-  getDepositableNodes = () => {
-    const { depositNodes } = this.props;
-    const nodes = Object.values(depositNodes.nodes);
-    const depositable = [];
-
-    for (let i = 0; i < nodes.length; i += 1) {
-      const node = nodes[i];
-      if (node.balanceSats.isEqualTo(0) && !node.addressUsed) {
-        depositable.push(node);
-      }
-    }
-    return depositable;
-  };
-
   getNextDepositAddress = () => {
     const { depositIndex } = this.state;
     this.setState({ depositIndex: depositIndex + 1 });
     setTimeout(this.getDepositAddress, 0);
   };
 
   getDepositAddress = () => {
-    const { network, client, updateDepositSlice } = this.props;
+    const {
+      network,
+      client,
+      updateDepositSlice,
+      depositableSlices,
+    } = this.props;
     const { depositIndex } = this.state;
-    const depositableNodes = this.getDepositableNodes();
-    if (depositIndex < depositableNodes.length)
+
+    if (depositIndex < depositableSlices.length)
       this.setState({
-        node: depositableNodes[depositIndex],
-        address: depositableNodes[depositIndex].multisig.address,
+        slice: depositableSlices[depositIndex],
+        address: depositableSlices[depositIndex].multisig.address,
         showReceived: false,
       });
 
     clearInterval(depositTimer);
     depositTimer = setInterval(async () => {
       let updates;
       try {
-        const { address } = this.state;
+        const { address, slice } = this.state;
         updates = await fetchAddressUTXOs(address, network, client);
         if (updates && updates.utxos && updates.utxos.length) {
           clearInterval(depositTimer);
-          updateDepositSlice(updates);
-          this.setState({ showReceived: true });
+          updateDepositSlice({ ...updates, bip32Path: slice.bip32Path });
+          this.setState({ showReceived: true }, () => {
+            this.resetDepositAddressIndex();
+            setTimeout(this.getDepositAddress, 2000);
```

Comment:
Needed a timeout since this method clears the alert informing the user that a deposit was received.